source/server-discovery-and-monitoring/tests/unified/logging-sharded.yml

```
@@ -0,0 +1,249 @@
+      - client: *client
+        ignoreMessages:
+          - level: debug
+            component: topology
+            data:
+              message: "Starting server monitoring"
+          - level: debug
+            component: topology
+            data:
+              message: "Server heartbeat started"
+          - level: debug
+            component: topology
+            data:
+              message: "Server heartbeat succeeded"
+          - level: debug
+            component: topology
+            data:
+              message: "Server heartbeat failed"
+        messages:
+          - level: debug
+            component: topology
+            data:
+              message: "Starting topology monitoring"
+              topologyId: { $$exists: true }
+          - level: debug
+            component: topology
+            data:
+              message: "Topology description changed"
+              topologyId: { $$exists: true }
+              previousDescription:  { $$exists: true } # unknown topology
+              newDescription: { $$exists: true } # Sharded topology
+          - level: debug
+            component: topology
+            data:
+              message: "Topology description changed"
+              topologyId: { $$exists: true }
+              previousDescription:  { $$exists: true }
+              newDescription: { $$exists: true } # shard router connected
+          - level: debug
+            component: topology
+            data:
+              message: "Topology description changed"
+              topologyId: { $$exists: true }
+              previousDescription:  { $$exists: true }
+              newDescription: { $$exists: true } # shard router connected
+          - level: debug
+            component: topology
+            data:
+              message: "Stopped server monitoring"
+              topologyId: { $$exists: true }
+              serverHost: { $$type: string }
+              serverPort: { $$type: [int, long] }
+          - level: debug
+            component: topology
+            data:
+              message: "Stopped server monitoring"
+              topologyId: { $$exists: true }
+              serverHost: { $$type: string }
+              serverPort: { $$type: [int, long] }
+          - level: debug
+            component: topology
+            data:
+              message: "Topology description changed"
+              topologyId: { $$exists: true }
+              previousDescription:  { $$exists: true } # Sharded topology
+              newDescription: { $$exists: true }  # unknown topology
+          - level: debug
+            component: topology
+            data:
+              message: "Stopped topology monitoring"
+              topologyId: { $$exists: true }
+  - description: Successful heartbeat
+    operations:
+      - name: createEntities
+        object: testRunner
+        arguments:
+          entities:
+          - client:
+              id: *client
+              observeLogMessages:
```

```suggestion
              uriOptions:
                  heartbeatFrequencyMS: 5000
              observeLogMessages:
```

If comment (A) is accepted, this would need the above udpate.